server/app/services/geo/esri/EsriClient.java

```diff
@@ -262,13 +270,41 @@ CompletionStage<Optional<JsonNode>> fetchServiceAreaFeatures(
 
   /**
    * Calls an external Esri service to get the service areas of the provided {@link
-   * AddressLocation}. Takes the returned service areas and returns a boolean specifying if the
-   * service area is included.
-   *
-   * @return an optional boolean if successful, or an empty optional if the request fails.
+   * AddressLocation}. Takes the returned service areas and returns an immutable list of {@link
+   * EsriServiceAreaInclusion}, filtered by the services areas specified in the application config
+   * that have the same {@link EsriServiceAreaValidationOption} URL.
    */
-  public CompletionStage<Optional<Boolean>> isAddressLocationInServiceArea(
+  public CompletionStage<ImmutableList<EsriServiceAreaInclusion>> getServiceAreaInclusionGroup(
       EsriServiceAreaValidationOption esriServiceAreaValidationOption, AddressLocation location) {
+    EsriServiceAreaInclusion.Builder esriServiceAreaInclusionBuilder =
+        EsriServiceAreaInclusion.builder();
+    ImmutableList.Builder<EsriServiceAreaInclusion> inclusionListBuilder = ImmutableList.builder();
+
+    Optional<ImmutableMap<String, EsriServiceAreaValidationOption>> maybeOptionMap =
+        esriServiceAreaValidationConfig.getImmutableMap();
+
+    if (maybeOptionMap.isEmpty()) {
+      logger.error(
+          "Error calling EsriClient.getServiceAreaInclusionGroups. Error:"
+              + " EsriServiceAreaValidationConfig.getImmutableMap() returned empty.");
+      return supplyAsync(
+          () -> {
+            esriServiceAreaInclusionBuilder
+                .setServiceAreaId(esriServiceAreaValidationOption.getId())
+                .setState(EsriServiceAreaState.FAILED)
+                .setTimeStamp(Instant.now());
+            inclusionListBuilder.add(esriServiceAreaInclusionBuilder.build());
+            return inclusionListBuilder.build();
+          });
+    }
+
+    // Create a list of options with the same URL as the passed in EsriServiceAreaValidationOption.
+    // In most cases the same URL should provide all of the service areas.
+    ImmutableList<EsriServiceAreaValidationOption> optionList =
+        maybeOptionMap.get().values().stream()
+            .filter(option -> option.getUrl().equals(esriServiceAreaValidationOption.getUrl()))
+            .collect(ImmutableList.toImmutableList());
```

Comment:
**suggestion (blocking):** recommend moving this to `EsriServiceAreaValidationConfig`

Reply:
Moved. Very open to suggestions on the new method name `mapToListWithSameServiceAreaOptionUrl`.